src/http/client.c

```diff
@@ -782,11 +791,23 @@ int http_request(struct http_req **reqp, struct http_cli *cli, const char *met,
 			goto out;
 	}
 	else {
+#ifdef HAVE_INET6
+		struct sa tmp;
+#endif
 		err = dnsc_query(&req->dq, cli->dnsc, req->host,
 				 DNS_TYPE_A, DNS_CLASS_IN, true,
 				 query_handler, req);
-		if (err)
-			goto out;
```

Comment:
I think we can keep the `goto out`. If dnsc_query setup fails, something is badly misconfigured or no mem is available.